addons/Popochiu/Editor/MainDock/TabRoom.gd

```diff
@@ -228,3 +301,107 @@ func _open_resource_script() -> void:
 	main_dock.ei.select_file(prd.get_script().resource_path)
 	main_dock.ei.set_main_screen_editor('Script')
 	main_dock.ei.edit_resource(prd.get_script())
+
+
+# Removes the character from the room
+func _on_remove_character_pressed(row: PopochiuObjectRow) -> void:
+	main_dock.show_confirmation(
+		'Remove character in room',
+		'Are you sure you want to remove [b]%s[/b] from this room?' % row.name,
+		'',
+		Vector2(200, 120)
+	)
+	
+	_remove_dialog.connect(
+		'confirmed', self, '_on_remove_character_confirmed', [row]
+	)
+	_remove_dialog.connect(
+		'popup_hide', self, '_on_remove_dialog_hide'
+	)
+
+
+func _on_remove_dialog_hide() -> void:
+	_remove_dialog.call_deferred(
+		'disconnect',
+		'confirmed', self, '_on_remove_character_confirmed'
+	)
+	_remove_dialog.call_deferred(
+		'disconnect',
+		'popup_hide', self, '_on_remove_dialog_hide'
+	)
+
+
+func _on_remove_character_confirmed(row: PopochiuObjectRow) -> void:
+	_characters_in_room.erase(row.name)
+	opened_room.get_node('Characters').get_node(
+		'Character%s *' % row.name
+	).queue_free()
+	row.queue_free()
+	main_dock.ei.save_scene()
+
+
+# Fills the list to show after pressing "+ Add character to room" and listens
+# to selections to add the clicked character to the room
+func _on_add_character_pressed() -> void:
+	var characters_menu := _btn_add_character.get_popup()
+	characters_menu.clear()
+	
+	var idx := 0
+	for key in PopochiuResources.get_section_keys('characters'):
+		characters_menu.add_item(key, idx)
+		characters_menu.set_item_disabled(idx, key in _characters_in_room)
+		
+		idx += 1
+	
+	if not characters_menu.is_connected(
+		'id_pressed', self, '_on_character_seleced'
+	):
+		characters_menu.connect('id_pressed', self, '_on_character_seleced')
+
+
+# Adds the clicked character in the "+ Add character to room" menu to the
+# current room
+func _on_character_seleced(id: int) -> void:
+	var characters_menu := _btn_add_character.get_popup()
+	var char_name := characters_menu.get_item_text(
+		characters_menu.get_item_index(id)
+	)
+	var instance: PopochiuCharacter = load(
+		'res://popochiu/Characters/%s/Character%s.tscn' % [char_name, char_name]
+	).instance()
+	instance.name = 'Character%s *' % char_name
+	
+	opened_room.get_node('Characters').add_child(instance)
+	instance.owner = opened_room
+	instance.position = Vector2(
+		ProjectSettings.get_setting(PopochiuResources.DISPLAY_WIDTH),
+		ProjectSettings.get_setting(PopochiuResources.DISPLAY_HEIGHT)
+	) / 2.0
+	
+	_characters_in_room.append(char_name)
+	
+	# Create the row for the character
```

Comment:
This seems like duplicate code to me, there is something similar in TabRoom, where I left my other comment... isn't there a way to make this less redundant?

Reply:
I wanted to have a variable to quickly know which characters are in the room, but you're right, I could also read the list of child nodes in the group. I'll see if I can change it.

Reply:
Wait, that was not my point. It just seemed to me that the same code is in `TabRoom.gd` too. Maybe it can be put in a function and reused? Or I may have missed it's not the same code 100%. Will check.